src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/AdapterUtil.cs

```diff
@@ -750,16 +750,20 @@ internal static Version GetAssemblyVersion()
 
         private const string ONDEMAND_PREFIX = "-ondemand";
         private const string AZURE_SYNAPSE = "-ondemand.sql.azuresynapse.";
+        private const string FABRIC_DATAWAREHOUSE = ".datawarehouse.fabric.microsoft.com";
 
         internal static bool IsAzureSynapseOnDemandEndpoint(string dataSource)
         {
-            return IsEndpoint(dataSource, ONDEMAND_PREFIX) || dataSource.Contains(AZURE_SYNAPSE);
+            return IsEndpoint(dataSource, ONDEMAND_PREFIX) 
+                || dataSource.Contains(AZURE_SYNAPSE)
+                || dataSource.Contains(FABRIC_DATAWAREHOUSE);
         }
 
         internal static readonly string[] s_azureSqlServerEndpoints = { StringsHelper.GetString(Strings.AZURESQL_GenericEndpoint),
```

Comment:
There are quite a few endpoints we could add here, e.g. https://learn.microsoft.com/en-us/fabric/security/fabric-allow-list-urls#data-warehouse and yes, they need not be localized.

Reply:
I will look into that.

Reply:
@cheenamalhotra Are all the 7 endpoint listed in your link similar to Synapse serverless??

Reply:
They're Fabric DW endpoints - so yes, they're serverless and synapse (data warehouse).